File: Newton/Python3/Newton.py
def f(x):
	return x * x - 2

def fd(x,h):
	return (f(x+h/2)-f(x-h/2))/h

def Newton(x,h):
	return x - f(x)/fd(x,h)

print('x = ', end='')
try:
	x = float(input())
except:
	x = 0
print('eps = ', end='')
try:
	eps = float(input())
except:
	eps = 0.0000000001
if eps >= 1 or eps <= 0:
	eps = 0.0000000001
print('h = ', end='')
try:
	h = float(input())
except:
	h = 0.00001
if h > 1 or h <= 0:
	h = 0.00001;
print('count = ', end='')
try:
	cnt = int(input())
except:
	cnt = 1000
if cnt <= 0:
	cnt = 1000
print('')

for i in range(cnt):
	x = Newton(x,h)
	if abs(f(x)) < eps:
		break

print("x = %f" % x)
print("f(x) = %f\n" % f(x))

File: Newton/C/Newton.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include "calc.h"

double f(double x) {
	return x * x - 2;
}

int main(void) {
	int i,count;
	double x,h,eps;
	char str[100];
	
	printf("x = ");
	fgets(str,sizeof(str),stdin);
	x = atof(str);
	printf("eps = ");
	fgets(str,sizeof(str),stdin);
	eps = atof(str);
	if ( eps >= 1 || eps <= 0 ) eps = 0.0000000001;
	printf("h = ");
	fgets(str,sizeof(str),stdin);
	h = atof(str);
	if ( h > 1 || h <= 0) h = 0.00001;
	printf("count = ");
	fgets(str,sizeof(str),stdin);
	count = atoi(str);
	if (count <= 0) count = 1000;
	puts("");
	
	for (i=0; i<count; i++) {
		x = Newton(x,h,f);
		if (fabs(f(x)) < eps) break;
	}
	printf("x = %f\nf(x) = %f\n",x,f(x));
	puts("");
	
	return 0;
}